utils/hints.go

```diff
@@ -233,6 +243,13 @@ func GenerateHints(annotations mapstr.M, container, prefix string) mapstr.M {
 						if strings.HasPrefix(hintKey, container) {
 							// Split the key to get part[1] to be the hint
 							parts := strings.Split(hintKey, "/")
+							//We check whether the provided annotation follows the supported format and vocabulary. The check happens for annotations that start with co.elastic.hints
+							for _, checksupported := range allSupportedHints {
```

Comment:
Maybe create a new function to check if it is supported since you are repeating this code twice in this function. It is easier to read then

Reply:
Done! checkSupportedHints added